src/cascadia/TerminalSettingsModel/IInheritable.h

```diff
@@ -0,0 +1,156 @@
+/*++
+Copyright (c) Microsoft Corporation
+Licensed under the MIT license.
+
+Module Name:
+- IInheritable.h
+
+Abstract:
+- An interface allowing settings objects to inherit settings from a parent
+
+Author(s):
+- Carlos Zamora - October 2020
+
+--*/
+#pragma once
+
+namespace winrt::Microsoft::Terminal::Settings::Model::implementation
+{
+    template<typename T>
+    struct IInheritable
+    {
+    public:
+        T* CreateChild() const
+        {
+            auto child{ winrt::make_self<T>() };
+            child->_parent.attach(const_cast<T*>(static_cast<const T*>(this)));
+            return child.detach();
+        }
+
+    protected:
+        com_ptr<T> _parent{ nullptr };
+    };
+
+    // Nullable settings are settings that explicitly allow null to mean something

```

Comment:
`// This is like std::optional, but we can use it in inheritance to determine whether the user explicitly cleared it`